xinference/core/model.py

```diff
@@ -235,8 +271,22 @@ def is_vllm_backend(self) -> bool:
 
         return isinstance(self._model, VLLMModel)
 
-    def load(self):
+    def allow_batching(self) -> bool:
+        from ..model.llm.pytorch.core import PytorchChatModel
+
+        return (
+            XINFERENCE_TRANSFORMERS_ENABLE_BATCHING
+            and isinstance(self._model, PytorchChatModel)
+            and self._model.__class__.__name__ == PytorchChatModel.__name__
+        )
+
+    async def load(self):
         self._model.load()
+        if self.allow_batching():
+            await self._scheduler_ref.set_model(self._model)
```

Comment:
Maybe we can make the model as a init param of `SchedulerActor` and create the `SchedulerActor` here.

Reply:
I tried this method. And I found that if passing self._model as an init parameter to an actor init method. The model will be copied.

Reply:
I got it. I have optimized the actor call in the same process to be a normal function call, with no data serialization. But the `create_actor` API uses the old protocol which will serialize/deserialize the input arguments.

Reply:
The full call stack in `create_actor`, we can see it go through the socket channel.

```
  File "/Users/codingl2k1/Work/inference/xinference/core/worker.py", line 650, in launch_builtin_model
    model_ref = await xo.create_actor(
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/api.py", line 78, in create_actor
    return await ctx.create_actor(actor_cls, *args, uid=uid, address=address, **kwargs)
      ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/backends/context.py", line 141, in create_actor
    future = await self._call(address, create_actor_message, wait=False)
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/backends/context.py", line 77, in _call
    return await self._caller.call(
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/backends/core.py", line 181, in call
    return await self.call_with_client(client, message, wait)
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/backends/core.py", line 126, in call_with_client
    await client.send(message)
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/backends/communication/base.py", line 278, in send
    return await self.channel.send(message)
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/backends/communication/socket.py", line 77, in send
    buffers = await serializer.run()
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/serialization/aio.py", line 71, in run
    return await self._get_buffers()
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/xoscar/serialization/aio.py", line 40, in _get_buffers
    headers, buffers = await serialize_with_spawn(
    ^^^^^^^^^^^^^^^^^
  File "xoscar/serialization/core.pyx", line 795, in serialize_with_spawn
    result = _serialize_with_stack(
    ^^^^^^^^^^^^^^^^^
  File "xoscar/serialization/core.pyx", line 689, in xoscar.serialization.core._serialize_with_stack
    serialized, subs, final = _serial_single(
    ^^^^^^^^^^^^^^^^^
  File "xoscar/serialization/core.pyx", line 627, in xoscar.serialization.core._serial_single
    ret_serial = serializer.serial(obj, context)
    ^^^^^^^^^^^^^^^^^
  File "xoscar/serialization/core.pyx", line 285, in xoscar.serialization.core.PickleSerializer.serial
    return (), pickle_buffers(obj), True
    ^^^^^^^^^^^^^^^^^
  File "xoscar/serialization/core.pyx", line 244, in xoscar.serialization.core.pickle_buffers
    buffers[0] = cloudpickle.dumps(
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/cloudpickle/cloudpickle_fast.py", line 73, in dumps
    cp.dump(obj)
  File "/Users/codingl2k1/.pyenv/versions/3.11.6/lib/python3.11/site-packages/cloudpickle/cloudpickle_fast.py", line 632, in dump
    return Pickler.dump(self, obj)
    ^^^^^^^^^^^^^^^^^
  File "/Users/codingl2k1/Work/inference/xinference/model/llm/pytorch/core.py", line 462, in __reduce__
    raise Exception("Forbid serialization")
    ^^^^^^^^^^^^^^^^^
Exception: [address=localhost:60075, pid=57865] Forbid serialization
```